.github/workflows/update-nock-files.yml

```diff
@@ -49,7 +49,15 @@ jobs:
 
       - name: Check if anything has changed
         id: contains-changes
-        run: echo "result=$(git --no-pager diff --quiet -- tests/nocks.db || echo "yes")" >> $GITHUB_OUTPUT
+        run: |
+          sqlite3 tests/nocks.db .dump > tests/nocks.sql
+          git add tests/nocks.sql
+          cp tests/nocks.db tests/nocks.db.new
+          git checkout HEAD -- tests/nocks.db
```

Comment:
nit: Rather than the `cp`, you could use `git show` to dump the old content in a `/tmp` file without mutating the real `nocks.db`.

Reply:
I've considered it, and as I was not sure what would be the behavior of `git show` with binary files, I decided to do that `cp` dance instead. I don't feel strongly either way, happy to change it if you think it's better

Reply:
I'd expect it to be possible but I can't find the right syntax locally, so feel free to merge as-is.